auth-api/src/auth_api/services/org.py

```diff
@@ -227,65 +228,84 @@ def _create_payment_settings(
+                pay_request["paymentInfo"]["bankAccountNumber"] = payment_info.get("bankAccountNumber", None)
+                pay_request["padTosAcceptedBy"] = kwargs["user_context"].user_name
+            # invoke pay-api
+            token = RestService.get_service_account_token()
+            if is_new_org:
+                response = RestService.post(
+                    endpoint=f"{pay_url}/accounts", data=pay_request, token=token, raise_for_status=True
+                )
+            else:
+                response = RestService.put(
+                    endpoint=f"{pay_url}/accounts/{org_model.id}", data=pay_request, token=token, raise_for_status=True
+                )
 
-        if payment_account_status != PaymentAccountStatus.FAILED and payment_method:
-            payment_method_description = (
-                PaymentMethod(payment_method).name if payment_method in [item.value for item in PaymentMethod] else ""
-            )
-            ActivityLogPublisher.publish_activity(
-                Activity(
-                    org_model.id,
-                    ActivityAction.PAYMENT_INFO_CHANGE.value,
-                    name=org_model.name,
-                    value=payment_method_description,
+            if response.status_code == HTTPStatus.OK:
+                payment_account_status = PaymentAccountStatus.CREATED
+            elif response.status_code == HTTPStatus.ACCEPTED:
+                payment_account_status = PaymentAccountStatus.PENDING
+            elif response.status_code == HTTPStatus.CREATED:
+                payment_account_status = PaymentAccountStatus.FAILED
+            else:
+                payment_account_status = PaymentAccountStatus.FAILED
+                if response.json:
+                    error_payload = response.json()
+                    error_code = error_payload.get("error", "UNKNOWN_ERROR")
+                else:
+                    error_code = "UNKNOWN_ERROR"
+                logger.error(f"Account create payment Error: {response.text}")
+
+            if payment_account_status != PaymentAccountStatus.FAILED and payment_method:
+                payment_method_description = (
+                    PaymentMethod(payment_method).name
+                    if payment_method in [item.value for item in PaymentMethod]
+                    else ""
                 )
-            )
-        return payment_account_status
+                ActivityLogPublisher.publish_activity(
+                    Activity(
+                        org_model.id,
+                        ActivityAction.PAYMENT_INFO_CHANGE.value,
+                        name=org_model.name,
+                        value=payment_method_description,
+                    )
+                )
+            return payment_account_status, error_code
+
+        except HTTPError as http_error:
+            error_payload = http_error.response.json()
+            error_code = error_payload.get("error", Error.ACCOUNT_CREATION_FAILED_IN_PAY)
```

Comment:
need some code to determine if it was POST or PUT and return ACCOUNT_UPDATE_FAILED_IN_PAY or ACCOUNT_CREATION_FAILED_IN_PAY

Reply:
alternatively make the error generic for both

`PAYMENT_ACCOUNT_FAILED`